### PR TITLE
New_dockerize-checker_Lisovska

### PR DESCRIPTION
**Pull request by okslisovska**
Currently, the checker service is working with 'solution' collection which it creates and populates itself.
But when we have a more stable version of our database then it will work with the data stored in mongodb.

### CODE REVIEW

File: solution-checker/index.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        os.unlink(self.fp.name)    
+      
+
+class PythonSolutionChecker(BaseSolutionChecker):
+    def is_solution_ok(self):
+        process = subprocess.run(["python3", self.fp.name], stderr=subprocess.DEVNULL)
+        return not process.returncode
+
+            
+language_mapping = {"Python": PythonSolutionChecker}
```
Comment (vvelizariy) [inline solution-checker/index.py]: This should be declared as constant and moved to top

File: solution-checker/index.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        
+    def is_solution_ok(self):
+        raise NotImplementedError
+
+
+    def remove_temp_file(self):
+        os.unlink(self.fp.name)    
+      
+
+class PythonSolutionChecker(BaseSolutionChecker):
```
Comment (vvelizariy) [inline solution-checker/index.py]: Docstring would be a plus)

File: solution-checker/index.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+            
+language_mapping = {"Python": PythonSolutionChecker}
+
+
+class TestRunnerDaemon:
+    def __init__(self, db):
+        self.db = db
+
+
```
Comment (vvelizariy) [inline solution-checker/index.py]: Please format according PEP8

File: solution-checker/index.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            s = db.solution.find_one({"status": "edited"})
+            if s:
+                checker = language_mapping[s["language"]](s["solution"], s["test_cases"]) 
+                result = checker.is_solution_ok() 
+                checker.remove_temp_file()
+                status = "correct" if result else "failed"    
+                print(status)
+                db.solution.update_one(
+                    {"_id": s["_id"]},
+                    {
```
Comment (vvelizariy) [inline solution-checker/index.py]: May be formatted as one line

File: solution-checker/index.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                checker.remove_temp_file()
+                status = "correct" if result else "failed"    
+                print(status)
+                db.solution.update_one(
+                    {"_id": s["_id"]},
+                    { 
+                        "$set": {"status": status}
+                    }
+                )
+            time.sleep(1)
```
Comment (vvelizariy) [inline solution-checker/index.py]: Is sleep required?

File: docker-compose.yaml [hunk 6]
```diff
@@ -46,6 +46,16 @@ services:
     depends_on:
       - api
 
+  checker:
```
Comment (vvelizariy) [inline docker-compose.yaml:52]: We should also block outgoing network requests for this container.
Example here https://stackoverflow.com/questions/39913757/restrict-internet-access-docker-container

**Review by DmytroBaryshniuk — APPROVED** [review]

File: solution-checker/index.py (hunk 1 above)
Comment (okslisovska) [reply solution-checker/index.py]: Uppercase was used to declare as const.
Moved to the top.

File: solution-checker/index.py (hunk 2 above)
Comment (okslisovska) [reply solution-checker/index.py]: Docstrings were added.

File: solution-checker/index.py (hunk 3 above)
Comment (okslisovska) [reply solution-checker/index.py]: Unnecessary blank lines were removed.

File: solution-checker/index.py (hunk 4 above)
Comment (okslisovska) [reply solution-checker/index.py]: Formatted in one line.
But now the line length exceeds 79
So which one is better? )

File: solution-checker/index.py (hunk 5 above)
Comment (okslisovska) [reply solution-checker/index.py]: sleep() removed.

File: docker-compose.yaml (hunk 6 above)
Comment (okslisovska) [reply docker-compose.yaml:52]: "no-internet" network was added.

File: checker.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,10 @@
+FROM python:latest
+
+WORKDIR /usr/src/checker
+
+COPY ./solution-checker .
+
+RUN pip install --no-cache-dir -r requirements.txt
+
+CMD ["python3", "index.py"]
```
Comment (vvelizariy) [inline checker.Dockerfile]: I would advice name not index(as it is not js file), but rather checker_service.py

Comment (okslisovska) [reply checker.Dockerfile]: Renamed index.py -> checker_service.py

**Review by s0fko — APPROVED** [review]

**Review by alexandrv1111 — APPROVED** [review]

**Review by MrSancho42 — APPROVED** [review]

**Pull request merged**